veinminer-server/src/main/java/wtf/choco/veinminer/tool/ToolCategoryRegistry.java

```diff
@@ -55,11 +56,10 @@ public VeinMinerToolCategory get(@NotNull String id) {
      * @return the corresponding tool category, or null if no category contains the item
      */
     @Nullable
-    public VeinMinerToolCategory get(@NotNull ItemType itemType, @NotNull Predicate<VeinMinerToolCategory> categoryPredicate) {
+    private VeinMinerToolCategory get(@NotNull ItemType itemType, @Nullable String itemNbtValue, @NotNull Predicate<VeinMinerToolCategory> categoryPredicate) {
         VeinMinerToolCategory resultCategory = null;
-
         for (VeinMinerToolCategory category : categories.values()) {
-            if (category.containsItem(itemType) && (resultCategory == null || category.compareTo(resultCategory) > 1) && categoryPredicate.test(category)) {
+            if (category.containsItem(itemType) && !(category.getNBTValue() != null && !category.getNBTValue().equals(itemNbtValue)) && (resultCategory == null || category.compareTo(resultCategory) > 1) && categoryPredicate.test(category)) {
```

Comment:
I would rather the condition match the same styling as the one for the category priorities.
```diff
- && !(category.getNBTValue() != null && !category.getNBTValue().equals(itemNbtValue))
+ && (category.getNBTValue() == null || category.getNBTValue().equals(itemNbtValue))
```